lib/api/objectPutCopyPart.js

```diff
@@ -133,6 +133,22 @@ function objectPutCopyPart(authInfo, request, sourceBucket,
                         sourceLocationConstraintName =
                             sourceObjMD.location[0].dataStoreName;
                     }
+                    // return error when object is cold
+                    if (sourceObjMD.archive &&
+                        // Object is in cold backend
+                        (!sourceObjMD.archive.restoreRequestedAt ||
+                        // Object is being restored
+                        (sourceObjMD.archive.restoreRequestedAt && !sourceObjMD.archive.restoreCompletedAt))) {
```

Comment:
should we introduce a function for this? (e.g. `isObjecDatatInColdStorage`)

or possibly a helper which verifies both that data is present and object is not being transitioned and returns the appropriate error? (e.g. `VerifyColdObjectAvailable`) ?